src/main/scala/io/epiphanous/flinkrunner/flink/state/ListState.scala

```diff
@@ -0,0 +1,31 @@
+package io.epiphanous.flinkrunner.flink.state
+
+
+class ListState[T] extends Serializable {
```

Comment:
What do you think about using a type class approach instead to add `isEmpty` and `contains` methods to Flink's `ListState` class. For instance, something like this (untested) code:

```scala
package io.epiphanous.flinkrunner.util

import org.apache.flink.api.common.state.ListState

import scala.collection.JavaConverters._

object StateUtils {
  implicit class RichListState[T](listState: ListState[T]) {
    def _iterable: Iterable[T]        =
      Option(listState.get()).map(_.asScala).getOrElse(Iterable.empty[T])
    def isEmpty: Boolean              = _iterable.isEmpty
    def contains(element: T): Boolean = _iterable.exists(_ == element)
  }
}
```

This should then let you use `isEmpty` and `contains` directly on instances of flink's `ListState` class.

Reply:
Okay, side conversation + https://docs.scala-lang.org/overviews/core/implicit-classes.html will update.